### PR TITLE
P4C-DPDK - Support Non-zero arguments for default action

### PR DESCRIPTION
**Pull request by github-sajan**

### CODE REVIEW

**Review by usha1830 — APPROVED** [review]
Looks good to me. Some tests are failing. Please check

File: backends/dpdk/dpdkArch.cpp [hunk 1]
```diff
@@ -1255,6 +1255,22 @@ const IR::Node *CollectLocalVariables::postorder(IR::P4Parser *p) {
...
 }
 
+void DefActionValue::postorder(const IR::P4Table *t) {
+    auto default_action = t->properties->getProperty("default_action");
+    if (default_action != nullptr && default_action->value->is<IR::ExpressionValue>()) {
+        auto expr = default_action->value->to<IR::ExpressionValue>()->expression;
+        auto mi = P4::MethodInstance::resolve(expr->to<IR::MethodCallExpression>(),
+                refMap, typeMap);
+        BUG_CHECK(mi->is<P4::ActionCall>(),
+            "%1%: expected action in default_action", default_action);
```
Comment (psivanup) [inline backends/dpdk/dpdkArch.cpp:1265]: should it be "unexpected method call in default_action"?

File: backends/dpdk/spec.cpp [hunk 2]
```diff
@@ -322,8 +322,24 @@ std::ostream &IR::DpdkTable::toSpec(std::ostream &out) const {
...
+                               components.at(i)->to<IR::Constant>()->asUnsigned();
+                    out << default_action_paraList.parameters.at(i)->toString() << " ";
+                    out << "0x" << std::hex << val << " ";
+                }
+            }
+        }
     }
+    auto def = properties->getProperty("default_action");
+    if (def->isConstant)
+        out <<"const";
```
Comment (psivanup) [inline backends/dpdk/spec.cpp:342]: space required after "<<"?

**Review by psivanup — COMMENTED** [review]
Please check the comments.

File: backends/dpdk/dpdkArch.cpp (hunk 1 above)
Comment (psivanup) [reply backends/dpdk/dpdkArch.cpp:1265]: sajan: it is similar to other error codes.

**Review by psivanup — APPROVED** [review]
The changes look fine to me.

File: backends/dpdk/dpdkArch.cpp (hunk 1 above)
Comment (github-sajan) [reply backends/dpdk/dpdkArch.cpp:1265]: Error looks fine

**Pull request closed without merging**